packages/models-library/src/models_library/api_schemas_webserver/users_preferences.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any, TypeAlias
+
+from pydantic import BaseModel, Field
+
+from ..services_ui import WidgetType
+from ..user_preferences import PreferenceName, ValueType
+
+
+class UserPreference(BaseModel):
+    render_widget: bool = Field(default=..., alias="exposeInPreferences")
+    widget_type: WidgetType | None = Field(default=..., alias="widget")
+    display_label: str | None = Field(default=..., alias="label")
+    tooltip_message: str | None = Field(default=..., alias="description")
+
+    value_type: ValueType = Field(default=..., alias="type")
+    default_value: Any = Field(default=..., alias="defaultValue")
+    value: Any
```

Comment:
Concerning the frontend preferences, I think it would be much more flexible if you keep just a dictionary (json in the db). 
Exceptions to this rule:
- preferences that we actually also need in the backend

Advantages are:
- if the frontend needs new preferences, it can add them very quickly on its end

What do you think?
Also, I don't understand what these entries are.

Reply:
and actually you can even save them as binary as well.

The point with the binary was for the services preferences where we do not have any control. the frontend ones we do.

Reply:
1. with OM we decided to drop all the extras

Reply:
this is out of date now.